src/NanopassSharp/Patterns/TransformationPatternExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NanopassSharp.Descriptions;
+
+namespace NanopassSharp.Patterns;
+
+public static class TransformationPatternExtensions
+{
+    /// <summary>
+    /// Returns a new <see cref="ITransformationPattern"/> which applies a binary and operation to two patterns
+    /// (<c><paramref name="left"/> &amp;&amp; <paramref name="right"/></c>).
+    /// </summary>
+    /// <param name="left">The left pattern.</param>
+    /// <param name="right">The right pattern.</param>
+    public static ITransformationPattern And(this ITransformationPattern left, ITransformationPattern right) =>
+        new BinaryPattern(left, right, (a, b) => a && b);
+
+    /// <summary>
+    /// Returns a new <see cref="ITransformationPattern"/> which applies a binary or operation to two patterns
+    /// (<c><paramref name="left"/> || <paramref name="right"/></c>).
+    /// </summary>
+    /// <param name="left">The left pattern.</param>
+    /// <param name="right">The right pattern.</param>
+    public static ITransformationPattern Or(this ITransformationPattern left, ITransformationPattern right) =>
+        new BinaryPattern(left, right, (a, b) => a || b);
+
+    /// <summary>
+    /// Returns a new <see cref="ITransformationPattern"/> which applies a binary exclusive or operation to two patterns
+    /// (<c><paramref name="left"/> ^ <paramref name="right"/></c>).
+    /// </summary>
+    /// <param name="left">The left pattern.</param>
+    /// <param name="right">The right pattern.</param>
+    public static ITransformationPattern Xor(this ITransformationPattern left, ITransformationPattern right) =>
+        new BinaryPattern(left, right, (a, b) => a ^ b);
+
+    /// <summary>
+    /// Returns a new <see cref="ITransformationPattern"/> which applies a unary not operation to a pattern.
+    /// (<c>!<paramref name="operand"/></c>).
+    /// </summary>
+    /// <param name="operand">The operand.</param>
+    public static ITransformationPattern Not(this ITransformationPattern operand) =>
+        new UnaryPattern(operand, x => !x);
+
+    /// <summary>
+    /// Returns a new <see cref="ITransformationPattern"/> which returns true
+    /// if any of the patterns in the collection return true.
+    /// </summary>
+    public static ITransformationPattern AnyPattern(this IEnumerable<ITransformationPattern> patterns) =>
```

Comment:
Any reason for the `Pattern` suffix? Are you afraid of name collision with LINQ?

My 2 cents here: renaming the class to something like `Patterns` would make this pretty simple to deal with, as people would type `Patterns.Any` and friends if it's ambiguous.